### PR TITLE
feat(jax/array-api): energy fitting

### PR DESCRIPTION
**Pull request by njzjz**
## Release Notes

- **New Features**
	- Introduced a fitting module for energy models using JAX, enhancing compatibility with different array backends.
	- Added `AtomExcludeMask` class for improved attribute handling in exclusion masks.

- **Improvements**
	- Updated serialization and array handling methods for better integration with array APIs.
	- Enhanced testing capabilities for energy fitting with support for different backends.

- **Documentation**
	- Added SPDX license identifier to relevant files for licensing clarity.

### CODE REVIEW

File: deepmd/jax/fitting/fitting.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    elif name == "nets":
+        value = NetworkCollection.deserialize(value.serialize())
+    return value
+
+
+@flax_module
+class EnergyFittingNet(EnergyFittingNetDP):
+    def __setattr__(self, name: str, value: Any) -> None:
+        value = setattr_for_general_fitting(name, value)
+        return super().__setattr__(name, value)
```
[the remark this answers is not in the document]
Comment (njzjz) [reply deepmd/jax/fitting/fitting.py:39]: It looks to me `-> None` is incorrect...

File: deepmd/dpmodel/fitting/general_fitting.py [hunk 2]
```diff
@@ -371,13 +376,15 @@ def _call_common(
                     "which is not consistent with {self.numb_fparam}.",
                 )
             fparam = (fparam - self.fparam_avg) * self.fparam_inv_std
-            fparam = np.tile(fparam.reshape([nf, 1, self.numb_fparam]), [1, nloc, 1])
-            xx = np.concatenate(
+            fparam = xp.tile(
+                xp.reshape(fparam, [nf, 1, self.numb_fparam]), (1, nloc, 1)
+            )
+            xx = xp.concat(
```
[the remark this answers is not in the document]
Comment (njzjz) [reply deepmd/dpmodel/fitting/general_fitting.py:382]: `xp.concat` is correct in the Array API standard. `xp.concatenate` is wrong.

File: deepmd/dpmodel/fitting/general_fitting.py [hunk 3]
```diff
@@ -415,15 +422,18 @@ def _call_common(
                 ):
                     assert xx_zeros is not None
                     atom_property -= self.nets[(type_i,)](xx_zeros)
-                atom_property = atom_property + self.bias_atom_e[type_i]
-                atom_property = atom_property * mask
+                atom_property = atom_property + self.bias_atom_e[type_i, ...]
+                atom_property = atom_property * xp.astype(mask, atom_property.dtype)
```
[the remark this answers is not in the document]
Comment (njzjz) [reply deepmd/dpmodel/fitting/general_fitting.py:426]: This is not supported by Array API.

File: deepmd/jax/fitting/fitting.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (njzjz) [reply deepmd/jax/fitting/fitting.py:39]: Oh, it returns None. But I don't think `return` is wrong.

**Review by wanghan-iapcm — APPROVED** [review]

**Review by anyangml — APPROVED** [review]

**Review by iProzd — APPROVED** [review]

**Pull request merged**